client/python/gradio_client/client.py

```diff
@@ -1129,24 +1130,47 @@ def _gather_files(self, *data):
         def get_file(d):
             if utils.is_file_obj(d):
                 file_list.append(d["path"])
+            elif isinstance(d, File):
+                file_list.append(d.path)
             else:
+                warnings.warn(
+                    f"The Client is treating: {d} as a filepath/URL and uploading it to the remote Gradio app. "
```

Comment:
I think we should keep the warning (but maybe throw it somewhere else) since URLs will also need be wrapped with `File()` (or whatever we call it) starting in 1.0